src/inmanta/moduletool.py

```diff
@@ -290,9 +301,20 @@ def bump_version(cls, change_type: ChangeType, version: Version, version_tag: st
             parts[0] += 1
             parts[1] = 0
             parts[2] = 0
+
+        if len(parts) >= 4:
+            size_version = 4
+
+            if change_type is ChangeType.REVISION:
```

Comment:
This doesn't allow us to do a revision on a 3-digit version, does it? I don't see that case in your tests either. I think we  do need to support it. It might be easier to make `parts` be size 4 always, and at the end of the method drop the fourth digit if it's a 0?

Open question: if you bump `1.2.3.4` with a patch, do we want to go to `1.2.4.0` or `1.2.4`? We should also add a test case for this.